### ISSUE TITLE
ppm not displaying

### ISSUE THREAD

**Issue by baentsch** [daniel-karl/covid19-map#65]
The ppm numbers are still displayed in the highscore but no longer in the map :-( A simple issue for you, @daniel-karl  or should I "dig"?

**Comment by daniel-karl** [comment]
Sorry for the long response time! It appears the populations for some newly added locations are missing, which breaks the PPM display altogether. For example, for Germany and Italy now individual regions are reported. There might be some other newly added location, a dump of all locations for which no population is registered would help...

**Comment by daniel-karl** [comment]
```["Abruzzo, Italy","Acre, Brazil","Adygea Republic, Russia","Aguascalientes, Mexico","Aichi, Japan","Akita, Japan","Alagoas, Brazil","Altai Krai, Russia","Altai Republic, Russia","Amapa, Brazil","Amazonas, Brazil","Amazonas, Colombia","Amazonas, Peru","Amur Oblast, Russia","Ancash, Peru","Andalusia, Spain","Antioquia, Colombia","Antofagasta, Chile","Aomori, Japan","Apurimac, Peru","Aragon, Spain","Arauca, Colombia","Araucania, Chile","Arequipa, Peru","Arica y Parinacota, Chile","Arkhangelsk Oblast, Russia","Astrakhan Oblast, Russia","Asturias, Spain","Atacama, Chile","Atlantico, Colombia","Ayacucho, Peru","Aysen, Chile","Baden-Wurttemberg, Germany","Bahia, Brazil","Baja California, Mexico","Baja California Sur, Mexico","Baleares, Spain","Bashkortostan Republic, Russia","Basilicata, Italy","Bayern, Germany","Belgorod Oblast, Russia","Berlin, Germany","Biobio, Chile","Bolivar, Colombia","Boyaca, Colombia","Brandenburg, Germany","Bremen, Germany","Bryansk Oblast, Russia","Buryatia Republic, Russia","C. Valenciana, Spain","Cajamarca, Peru","Calabria, Italy","Caldas, Colombia","Callao, Peru","Campania, Italy","Campeche, Mexico","Canarias, Spain","Cantabria, Spain","Capital District, Colombia","Caqueta, Colombia","Casanare, Colombia","Castilla - La Mancha, Spain","Castilla y Leon, Spain","Catalonia, Spain","Cauca, Colombia","Ceara, Brazil","Cesar, Colombia","Ceuta, Spain","Chechen Republic, Russia","Chelyabinsk Oblast, Russia","Cherkasy Oblast, Ukraine","Chernihiv Oblast, Ukraine","Chernivtsi Oblast, Ukraine","Chiapas, Mexico","Chiba, Japan","Chihuahua, Mexico","Choco, Colombia","Chukotka Autonomous Okrug, Russia","Chuvashia Republic, Russia","Ciudad de Mexico, Mexico","Coahuila, Mexico","Colima, Mexico","Coquimbo, Chile","Cordoba, Colombia","Crimea Republic*, Ukraine","Cundinamarca, Colombia","Cusco, Peru","Dagestan Republic, Russia","Distrito Federal, Brazil","Dnipropetrovsk Oblast, Ukraine","Donetsk Oblast, Ukraine","Durango, Mexico","Ehime, Japan","Emilia-Romagna, Italy","Espirito Santo, Brazil","Extremadura, Spain","Friuli Venezia Giulia, Italy","Fukui, Japan","Fukuoka, Japan","Fukushima, Japan","Galicia, Spain","Gifu, Japan","Goias, Brazil","Guainia, Colombia","Guanajuato, Mexico","Guerrero, Mexico","Gunma, Japan","Hamburg, Germany","Hessen, Germany","Hidalgo, Mexico","Hiroshima, Japan","Hokkaido, Japan","Huancavelica, Peru","Huanuco, Peru","Huila, Colombia","Hyogo, Japan","Ibaraki, Japan","Ica, Peru","Ingushetia Republic, Russia","Irkutsk Oblast, Russia","Ishikawa, Japan","Ivano-Frankivsk Oblast, Ukraine","Ivanovo Oblast, Russia","Jalisco, Mexico","Jewish Autonomous Okrug, Russia","Junin, Peru","Kabardino-Balkarian Republic, Russia","Kagawa, Japan","Kagoshima, Japan","Kaliningrad Oblast, Russia","Kalmykia Republic, Russia","Kaluga Oblast, Russia","Kamchatka Krai, Russia","Kanagawa, Japan","Karachay-Cherkess Republic, Russia","Karelia Republic, Russia","Kemerovo Oblast, Russia","Khabarovsk Krai, Russia","Khakassia Republic, Russia","Khanty-Mansi Autonomous Okrug, Russia","Kharkiv Oblast, Ukraine","Kherson Oblast, Ukraine","Khmelnytskyi Oblast, Ukraine","Kiev, Ukraine","Kiev Oblast, Ukraine","Kirov Oblast, Russia","Kirovohrad Oblast, Ukraine","Kochi, Japan","Komi Republic, Russia","Kostroma Oblast, Russia","Krasnodar Krai, Russia","Krasnoyarsk Krai, Russia","Kumamoto, Japan","Kurgan Oblast, Russia","Kursk Oblast, Russia","Kyoto, Japan","La Guajira, Colombia","La Libertad, Peru","La Rioja, Spain","Lambayeque, Peru","Lazio, Italy","Leningrad Oblast, Russia","Liguria, Italy","Lima, Peru","Lipetsk Oblast, Russia","Lombardia, Italy","Loreto, Peru","Los Lagos, Chile","Los Rios, Chile","Luhansk Oblast, Ukraine","Lviv Oblast, Ukraine","Madre de Dios, Peru","Madrid, Spain","Magadan Oblast, Russia","Magallanes, Chile","Magdalena, Colombia","Maranhao, Brazil","Marche, Italy","Mari El Republic, Russia","Mato Grosso, Brazil","Mato Grosso do Sul, Brazil","Maule, Chile","Mecklenburg-Vorpommern, Germany","Melilla, Spain","Meta, Colombia","Metropolitana, Chile","Mexico, Mexico","Michoacan, Mexico","Mie, Japan","Minas Gerais, Brazil","Miyagi, Japan","Miyazaki, Japan","Molise, Italy","Moquegua, Peru","Mordovia Republic, Russia","Morelos, Mexico","Moscow, Russia","Moscow Oblast, Russia","Murcia, Spain","Murmansk Oblast, Russia","Mykolaiv Oblast, Ukraine","Nagano, Japan","Nagasaki, Japan","Nara, Japan","Narino, Colombia","Navarra, Spain","Nayarit, Mexico","Nenets Autonomous Okrug, Russia","Niedersachsen, Germany","Niigata, Japan","Nizhny Novgorod Oblast, Russia","Nordrhein-Westfalen, Germany","Norte de Santander, Colombia","North Ossetia - Alania Republic, Russia","Novgorod Oblast, Russia","Novosibirsk Oblast, Russia","Nuble, Chile","Nuevo Leon, Mexico","OHiggins, Chile","Oaxaca, Mexico","Odessa Oblast, Ukraine","Oita, Japan","Okayama, Japan","Okinawa, Japan","Omsk Oblast, Russia","Orel Oblast, Russia","Orenburg Oblast, Russia","Osaka, Japan","P.A. Bolzano, Italy","P.A. Trento, Italy","Pais Vasco, Spain","Para, Brazil","Paraiba, Brazil","Parana, Brazil","Pasco, Peru","Penza Oblast, Russia","Perm Krai, Russia","Pernambuco, Brazil","Piaui, Brazil","Piemonte, Italy","Piura, Peru","Poltava Oblast, Ukraine","Port Quarantine, Japan","Primorsky Krai, Russia","Pskov Oblast, Russia","Puebla, Mexico","Puglia, Italy","Puno, Peru","Putumayo, Colombia","Queretaro, Mexico","Quindio, Colombia","Quintana Roo, Mexico","Rheinland-Pfalz, Germany","Rio Grande do Norte, Brazil","Rio Grande do Sul, Brazil","Rio de Janeiro, Brazil","Risaralda, Colombia","Rivne Oblast, Ukraine","Rondonia, Brazil","Roraima, Brazil","Rostov Oblast, Russia","Ryazan Oblast, Russia","Saarland, Germany","Sachsen, Germany","Sachsen-Anhalt, Germany","Saga, Japan","Saint Petersburg, Russia","Saitama, Japan","Sakha (Yakutiya) Republic, Russia","Sakhalin Oblast, Russia","Samara Oblast, Russia","San Andres y Providencia, Colombia","San Luis Potosi, Mexico","San Martin, Peru","Santa Catarina, Brazil","Santander, Colombia","Sao Paulo, Brazil","Saratov Oblast, Russia","Sardegna, Italy","Schleswig-Holstein, Germany","Sergipe, Brazil","Sevastopol*, Ukraine","Shiga, Japan","Shimane, Japan","Shizuoka, Japan","Sicilia, Italy","Sinaloa, Mexico","Smolensk Oblast, Russia","Sonora, Mexico","Stavropol Krai, Russia","Sucre, Colombia","Sumy Oblast, Ukraine","Sverdlovsk Oblast, Russia","Tabasco, Mexico","Tacna, Peru","Tamaulipas, Mexico","Tambov Oblast, Russia","Tarapaca, Chile","Tatarstan Republic, Russia","Ternopil Oblast, Ukraine","Thuringen, Germany","Tlaxcala, Mexico","Tocantins, Brazil","Tochigi, Japan","Tokushima, Japan","Tokyo, Japan","Tolima, Colombia","Tomsk Oblast, Russia","Toscana, Italy","Tottori, Japan","Toyama, Japan","Tula Oblast, Russia","Tumbes, Peru","Tver Oblast, Russia","Tyumen Oblast, Russia","Tyva Republic, Russia","Ucayali, Peru","Udmurt Republic, Russia","Ulyanovsk Oblast, Russia","Umbria, Italy","Unknown, Brazil","Unknown, Chile","Unknown, Colombia","Unknown, Germany","Unknown, Japan","Unknown, Peru","Unknown, Spain","Valle d'Aosta, Italy","Valle del Cauca, Colombia","Valparaiso, Chile","Vaupes, Colombia","Veneto, Italy","Veracruz, Mexico","Vichada, Colombia","Vinnytsia Oblast, Ukraine","Vladimir Oblast, Russia","Volgograd Oblast, Russia","Vologda Oblast, Russia","Volyn Oblast, Ukraine","Voronezh Oblast, Russia","Wakayama, Japan","Yamagata, Japan","Yamaguchi, Japan","Yamalo-Nenets Autonomous Okrug, Russia","Yamanashi, Japan","Yaroslavl Oblast, Russia","Yucatan, Mexico","Zabaykalsky Krai, Russia","Zacatecas, Mexico","Zakarpattia Oblast, Ukraine","Zaporizhia Oblast, Ukraine","Zhytomyr Oblast, Ukraine"]```

**Comment by daniel-karl** [comment]
for each of these an entry in `Population.js` would be required

**Comment by daniel-karl** [comment]
In order to reproduce the above list put 

```
let missing = [];
                                        Object.keys(ds.locations).map((location) => {
                                           if(!Object.keys(Population.ABSOLUTE).includes(location)) {
                                               missing.push(location);
                                           }
                                        });
                                        console.log(JSON.stringify(missing));
```

in `JHDataSourceProvider.js:61` and start the application

**Comment by baentsch** [comment]
@daniel-karl Thanks for the list. If I had time at my hands, I'd try to collect these numbers, but a) I don't and b) locations like ""Unknown, Brazil"" make it look like I cannot be 100% successful which seems necessary to get any display back. So may I ask whether it's not easier to catch/ignore the error if no population number is given (and not display 'ppm-visuals' in (only) those cases -- but display all the rest)?

**Comment by daniel-karl** [comment]
Fixed in 8cc4840
